src/main/java/org/primefaces/component/datatable/DataTable.java

```diff
@@ -141,25 +141,26 @@
     public static final String SMALL_SIZE_CLASS = "ui-datatable-sm";
     public static final String LARGE_SIZE_CLASS = "ui-datatable-lg";
 
-    public static final Map<DataTableFeatureKey, DataTableFeature> FEATURES = MapBuilder.<DataTableFeatureKey, DataTableFeature>builder()
-            .put(DataTableFeatureKey.DRAGGABLE_COLUMNS, new DraggableColumnsFeature())
-            .put(DataTableFeatureKey.FILTER, new FilterFeature())
+    public static final Map<DataTableFeatureKey, DataTableFeature> FEATURES = MapBuilder.<DataTableFeatureKey, DataTableFeature>builder( new LinkedHashMap<>() )
             .put(DataTableFeatureKey.PAGE, new PageFeature())
+            .put(DataTableFeatureKey.SCROLL, new ScrollFeature())
+            .put(DataTableFeatureKey.FILTER, new FilterFeature())
             .put(DataTableFeatureKey.SORT, new SortFeature())
-            .put(DataTableFeatureKey.RESIZABLE_COLUMNS, new ResizableColumnsFeature())
             .put(DataTableFeatureKey.SELECT, new SelectionFeature())
+            .put(DataTableFeatureKey.DRAGGABLE_COLUMNS, new DraggableColumnsFeature())
+            .put(DataTableFeatureKey.RESIZABLE_COLUMNS, new ResizableColumnsFeature())
             .put(DataTableFeatureKey.ROW_EDIT, new RowEditFeature())
             .put(DataTableFeatureKey.CELL_EDIT, new CellEditFeature())
             .put(DataTableFeatureKey.ROW_EXPAND, new RowExpandFeature())
-            .put(DataTableFeatureKey.SCROLL, new ScrollFeature())
             .put(DataTableFeatureKey.DRAGGABLE_ROWS, new DraggableRowsFeature())
             .put(DataTableFeatureKey.ADD_ROW, new AddRowFeature())
             .build();
 
-    private static final Map<String, Class<? extends BehaviorEvent>> BEHAVIOR_EVENT_MAPPING = MapBuilder.<String, Class<? extends BehaviorEvent>>builder()
+    private static final Map<String, Class<? extends BehaviorEvent>> BEHAVIOR_EVENT_MAPPING =
+        MapBuilder.<String, Class<? extends BehaviorEvent>>builder( new LinkedHashMap<>() )
```

Comment:
I don't think order matters here, does it?

Reply:
actually not necessary for the fix. the idea is to get a predictable order for processing features and to process features that might modify table data first.

Reply:
@robtha I think Rapster is saying the LinkedhashMap is good for the Features but not necessary for BehaviorEvents.

Reply:
@melloware I have to admit that I didn't really check if the order of BehaviourEventS has any impact.